### PR TITLE
Fix i519

### PR DESCRIPTION
**Pull request by KacperFKorban**
closes zio#519

/claim #519

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -145,6 +145,15 @@ lazy val zioSchemaDerivation = crossProject(JSPlatform, JVMPlatform)
...
   )
+  .settings(
+    Test / scalacOptions ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((3, _)) =>
+          Seq("-Wconf:name=MatchCaseOnlyNullWarning:e")
+        case _ => Seq.empty
+      }
+    }
+  )
```
Comment (tusharmath) [inline build.sbt:156]: Might want to remove the `scalacOptions -= "-Xfatal-warnings"` instruction that's added at a few places in the `build.sbt` file

**Comment by jdegoes** [comment]
@tusharmath Is this good to merge?

File: build.sbt (hunk 1 above)
Comment (KacperFKorban) [reply build.sbt]: Ok, removed this custom setting and removed `dottySettings` from `zioSchemaDerivation` instead.

**Comment by tusharmath** [comment]
No, it still fails with the following error —

```
[error] -- [E121] Pattern Match Error: /Users/tushar/Documents/Projects/zio-schema/zio-schema-examples/shared/src/main/scala/dev/zio/schema/example/Foo.scala:8:53 
[error]   8 |  implicit val schema: Schema[Foo] = DeriveSchema.gen[Foo]
[error]     |                                     ^^^^^^^^^^^^^^^^^^^^^
[error]     |Unreachable case except for null (if this is intentional, consider writing case null => instead).
[error]     |---------------------------------------------------------------------------
[error]     |Inline stack trace
[error]     |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]     |This location contains code that was inlined from DeriveSchema.scala:461
[error] 461 |          case other => throw new MatchError(other)
[error]     |               ^^^^^
[error]     |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]     |This location contains code that was inlined from DeriveSchema.scala:461
[error] 458 |      val unsafeDeconstruct = '{ 
[error]     |                              ^
[error] 459 |        (z: T) => z match {
[error] 460 |          case (sub: t) => sub
[error] 461 |          case other => throw new MatchError(other)
[error] 462 |        }
[error] 463 |       }
[error]      ---------------------------------------------------------------------------
[error] one error found

```


**[repro.patch](https://github.com/zio/zio-schema/files/11339511/repro.patch)**

- Apply the attached patch to your revision.
- Run the following command `reload; clean; ++3.2.1;zioSchemaExamplesJVM/compile`
- **Expected** — code should compile
- **Actual** - code doesn't compile.

**Comment by tusharmath** [comment]
Good to merge!

**Review by tusharmath — APPROVED** [review]

**Pull request merged**